offset_manager.go

```diff
@@ -250,8 +251,12 @@ func (pom *partitionOffsetManager) fetchInitialOffset(retries int) error {
 
 	switch block.Err {
 	case ErrNoError:
-		pom.offset = block.Offset
-		pom.metadata = block.Metadata
+		if block.Offset >= 0 {
+			pom.offset = block.Offset
+			pom.metadata = block.Metadata
+		} else {
+			pom.offset = pom.parent.conf.Consumer.Offsets.Initial
```

Comment:
I prefer returning the constant. This way to user of the API can know there's no offset stored, and do something specific in that case (e.g. log). Moreover, in the end you feed to value to `ConsumePartition`, which will always do a `FetchOffset` API call to validate/lookup the actual offset to use. So this saves you one roundtrip to the broker.